dynamics/client.py

```diff
@@ -181,9 +198,14 @@ def from_environment(cls):
         token_url = os.environ["DYNAMICS_TOKEN_URL"]
         client_id = os.environ["DYNAMICS_CLIENT_ID"]
         client_secret = os.environ["DYNAMICS_CLIENT_SECRET"]
-        scope = os.environ["DYNAMICS_SCOPE"].split(",")
+        scope = os.environ.get("DYNAMICS_SCOPE")
+        if scope is None:
+            raise KeyError("DYNAMICS_SCOPE")
```

Comment:
Isn't `scope` now mandatory, as you'll need to set something to it or you'll get `None`? It should be optional, and KeyError raised only if both it and `resource` are missing.

Reply:
I'll implement your preference (still require `scope`, or "one-is-enough" from (`scope`, `resource`)

Reply:
Scope should be optional, see [this comment](https://github.com/MrThearMan/dynamics-client/pull/4/files#r1152125685).

Reply:
Done.